project/src/main/webapp/matchbrackets.js

```diff
@@ -0,0 +1,158 @@
+// CodeMirror, copyright (c) by Marijn Haverbeke and others
```

Comment:
Same here. At least add more comments to explain the hacks (there are more comments than the previous file, but can still use more. For instance I don't seem to see any explanation about how comments are denoted - with what special characters?).

Reply:
this is directly from code mirror, so should i still comment it?
https://codemirror.net/addon/edit/matchbrackets.js